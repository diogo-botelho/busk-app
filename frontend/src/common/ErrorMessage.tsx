import { Alert } from "react-bootstrap";

/** Presentational component for showing bootstrap-style alerts.
 *
 * { LoginForm, SignupForm, EventList  } -> ErrorMessage
 **/

interface ErrorMessageParams {
  messages: string[];
}

function ErrorMessage({ messages }: ErrorMessageParams) {
  return (
    <Alert variant="danger">
      <Alert.Heading>Oh snap! You got an error!</Alert.Heading>
      {messages.map((message, idx) => (
        <p key={idx}>{message}</p>
      ))}
    </Alert>
  );
}

export default ErrorMessage;
