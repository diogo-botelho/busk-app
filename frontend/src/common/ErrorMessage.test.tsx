import React from "react";
import { render } from "@testing-library/react";
import ErrorMessage from "./ErrorMessage";

it("renders without crashing", function() {
  render(<ErrorMessage messages={[]}/>);
});

it("matches snapshot for danger", function() {
  let messages = ["Everything is broken", "Run for the hills"];
  const { asFragment } = render(<ErrorMessage messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot for success", function() {
  let messages = ["Everything is awesome!"];
  const { asFragment } = render(<ErrorMessage messages={messages} />);
  expect(asFragment()).toMatchSnapshot();
});
