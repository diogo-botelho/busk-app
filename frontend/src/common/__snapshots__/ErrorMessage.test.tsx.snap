// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for danger 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-danger show"
    role="alert"
  >
    <div
      class="alert-heading h4"
    >
      Oh snap! You got an error!
    </div>
    <p>
      Everything is broken
    </p>
    <p>
      Run for the hills
    </p>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for success 1`] = `
<DocumentFragment>
  <div
    class="fade alert alert-danger show"
    role="alert"
  >
    <div
      class="alert-heading h4"
    >
      Oh snap! You got an error!
    </div>
    <p>
      Everything is awesome!
    </p>
  </div>
</DocumentFragment>
`;
